addons/hr_timesheet/models/project_task.py

```diff
@@ -27,7 +27,11 @@ class Task(models.Model):
 
     project_id = fields.Many2one(domain="['|', ('company_id', '=', False), ('company_id', '=?',  company_id), ('is_internal_project', '=', False)]")
     analytic_account_active = fields.Boolean("Active Analytic Account", compute='_compute_analytic_account_active', compute_sudo=True, recursive=True)
-    allow_timesheets = fields.Boolean("Allow timesheets", related="project_id.allow_timesheets", help="Timesheets can be logged on this task.", readonly=True, recursive=True)
+    allow_timesheets = fields.Boolean(
+        "Allow timesheets",
+        compute='_compute_allow_timesheets', search='_search_allow_timesheets',
+        compute_sudo=True, readonly=True, recursive=True,
```

Comment:
```suggestion
        compute_sudo=True, readonly=True,
```
I think the recursive is no longer needed. 🤔

Reply:
Should I remove it in `17.0` as well, or is just `master` good?

Reply:
> Yes, I think you're right, I believe it was only useful with `project_root_id`?

Yep indeed 😀